### PR TITLE
Moving unique_name to python

### LINKED ISSUE #8526: Add API to reset unique_name generator

### PR DESCRIPTION
**Pull request by reyoung**
* Add reset and guard to unique_name

Fix #8526

### CODE REVIEW

File: python/paddle/v2/fluid/unique_name.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        self.prefix = prefix
+
+    def __call__(self, key):
+        """
+        Generate unique names with prefix
+
+        Args:
+            key(str): The key of return string.
+
+        Returns(str): A unique string with the prefix
```
Comment (jacquesqiao) [inline python/paddle/v2/fluid/unique_name.py]: seems prefix is not used in generated name

File: python/paddle/v2/fluid/unique_name.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        """
+        tmp = self.ids[key]
+        self.ids[key] += 1
+        return key + "_".join(key, str(tmp))
+
+
+generator = UniqueNameGenerator()
```
[the remark this answers is not in the document]
Comment (reyoung) [reply python/paddle/v2/fluid/unique_name.py]: The `generator` is a private member of unique_name.py . It is a little bit noising to add `g_`

File: python/paddle/v2/fluid/unique_name.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def generate(prefix):
```
[the remark this answers is not in the document]
Comment (reyoung) [reply python/paddle/v2/fluid/unique_name.py]: Done.

File: python/paddle/v2/fluid/unique_name.py (hunk 1 above)
Comment (reyoung) [reply python/paddle/v2/fluid/unique_name.py]: It is used by self.prefix

File: python/paddle/v2/fluid/unique_name.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+@contextlib.contextmanager
+def guard(new_generator=None):
+    old = switch(new_generator)
+    yield
+    switch(old)
+
+
+__call__ = generate
```
[the remark this answers is not in the document]
Comment (reyoung) [reply python/paddle/v2/fluid/unique_name.py]: Done.

File: python/paddle/v2/fluid/__init__.py [hunk 5]
```diff
@@ -39,30 +39,16 @@
...
-    'Tensor',
-    'ParamAttr',
-    'WeightNormParamAttr',
-    'DataFeeder',
-    'clip',
-    'SimpleDistributeTranspiler',
-    'DistributeTranspiler',
-    'memory_optimize',
-    'profiler',
+    'io', 'initializer', 'layers', 'nets', 'optimizer', 'learning_rate_decay',
```
Comment (jacquesqiao) [inline python/paddle/v2/fluid/__init__.py]: It's better to add a `,` after `'unique_name'`, because the format of this snippet code always changes.

**Pull request merged**